### ISSUE TITLE
LLVM abort with indexing of byte string literal in static

### ISSUE THREAD

**Issue by huonw** [rust-lang/rust#17233]
```rust
#![allow(dead_code)]

const X: &'static [u8] = b"1";

static Y: u8 = X[0];

fn main() {}
```

```
rustc: /build/rust-git/src/rust/src/llvm/include/llvm/Support/Casting.h:237: typename llvm::cast_retty<X, Y*>::ret_type llvm::cast(Y*) [with X = llvm::GlobalVariable; Y = llvm::Value; typename llvm::cast_retty<X, Y*>::ret_type = llvm::GlobalVariable*]: Assertion `isa<X>(Val) && "cast<Ty>() argument of incompatible type!"' failed.
Aborted
playpen: application terminated with error code 134
```

Changing the `b"1"` to e.g. `&[1]` works fine.

**Comment by vadimcn** [comment]
Still seeing this, but `bytes!()` is gone now :(

**Comment by huonw** [comment]
This seems to be related to `b"..."` being a DST, that seemingly isn't being handled correctly, cc @nick29581. I think it would be fixed by https://github.com/rust-lang/rust/issues/18465 but both @alexcrichton and I have met problems with implementing that.

**Comment by JustAPerson** [comment]
This works fine now.

This issue may be closed.

**Comment by nrc** [comment]
Closed by #22838 (which even included a test :-) )
